document-store/src/test/java/org/hypertrace/core/documentstore/postgres/query/v1/PostgresQueryParserTest.java

```diff
@@ -542,4 +543,112 @@ void testFindWithSortingAndPagination() {
     Assertions.assertEquals(1, params.getObjectParams().get(5));
     Assertions.assertEquals(3, params.getObjectParams().get(6));
   }
+
+  @Test
+  void testUnnestWithoutPreserveNullAndEmptyArrays() {
+    org.hypertrace.core.documentstore.query.Query query =
+        org.hypertrace.core.documentstore.query.Query.builder()
+            .addSelection(IdentifierExpression.of("item"))
+            .addSelection(IdentifierExpression.of("price"))
+            .addSelection(IdentifierExpression.of("sales.city"))
+            .addSelection(IdentifierExpression.of("sales.medium.type"))
+            .addFromClause(UnnestExpression.of(IdentifierExpression.of("sales"), false))
+            .addFromClause(UnnestExpression.of(IdentifierExpression.of("sales.medium"), false))
+            .build();
+
+    PostgresQueryParser postgresQueryParser = new PostgresQueryParser(TEST_COLLECTION, query);
+    String sql = postgresQueryParser.parse();
+
+    Assertions.assertEquals(
+        "With \n"
+            + "table1 as (SELECT * from testCollection), \n"
+            + "table2 as (SELECT * FROM table1, "
+            + "jsonb_array_elements(document->'sales') p1(sales),"
+            + "jsonb_array_elements(sales->'medium') p2(sales_dot_medium)) \n"
+            + "SELECT document->'item' AS item, document->'price' AS price, "
+            + "sales->'city' AS sales_dot_city, "
+            + "sales_dot_medium->'type' AS sales_dot_medium_dot_type "
+            + "FROM table2",
+        sql);
+
+    Params params = postgresQueryParser.getParamsBuilder().build();
+    Assertions.assertEquals(0, params.getObjectParams().size());
+  }
+
+  @Test
+  void testUnnestWithPreserveNullAndEmptyArrays() {
+    org.hypertrace.core.documentstore.query.Query query =
+        org.hypertrace.core.documentstore.query.Query.builder()
+            .addSelection(IdentifierExpression.of("item"))
+            .addSelection(IdentifierExpression.of("price"))
+            .addSelection(IdentifierExpression.of("sales.city"))
+            .addSelection(IdentifierExpression.of("sales.medium.type"))
+            .addFromClause(UnnestExpression.of(IdentifierExpression.of("sales"), true))
+            .addFromClause(UnnestExpression.of(IdentifierExpression.of("sales.medium"), true))
+            .build();
+
+    PostgresQueryParser postgresQueryParser = new PostgresQueryParser(TEST_COLLECTION, query);
+    String sql = postgresQueryParser.parse();
+
+    Assertions.assertEquals(
+        "With \n"
+            + "table1 as (SELECT * from testCollection), \n"
```

Comment:
table1 seems unused. Can we get rid of this temporary table?

Reply:
`table2 as (SELECT * FROM table1, jsonb_array_elements(document->'sales') p1(sales)`

Need it when there is a `WHERE` caluse as it should apply before unwind.